lambda/app/src/bcsc/client.ts

```diff
@@ -0,0 +1,100 @@
+import { models } from '../../../shared/sequelize/models/models';
+import axios from 'axios';
+import { IntegrationData } from '@lambda-shared/interfaces';
+import { getBCSCEnvVars } from '@lambda-app/utils/helpers';
+import { bcscDefaultScopes } from '@lambda-app/utils/constants';
+import { getAllEmailsOfTeam } from '@lambda-app/queries/team';
+
+export interface BCSCClientParameters {
+  id?: number;
+  clientId?: string;
+  clientName?: string;
+  clientUri?: string;
+  clientSecret?: string;
+  /** Provide scope as a space-separated string, e.g "openid address profile" */
+  contacts?: string[];
+  tokenEndpointAuthMethod?: string;
+  idTokenSignedResponseAlg?: string;
+  userinfoSignedResponseAlg?: string;
+  created?: boolean;
+  registrationAccessToken?: string;
+  environment?: string;
+}
+
+export const createBCSCClient = async (data: BCSCClientParameters, integration: IntegrationData, userId: number) => {
+  let contacts = [];
+  if (integration.usesTeam) {
+    const teamEmails = await getAllEmailsOfTeam(Number(integration.teamId));
+    contacts = teamEmails.map((member) => member.idir_email);
+  } else {
+    const contact = await models.user.findOne({
+      where: {
+        id: userId,
+      },
+    });
+    contacts.push(contact.idirEmail);
+  }
+  const { bcscBaseUrl, kcBaseUrl, accessToken } = getBCSCEnvVars(data.environment);
+  const jwksUri = `${kcBaseUrl}/realms/standard/protocol/openid-connect/certs`;
+
+  const result = await axios.post(
+    `${bcscBaseUrl}/oauth2/register`,
+    {
+      client_name: `${data.clientName}-${data.environment}`,
```

Comment:
@jlangy how important is to have environment in the client name? clients created at id.gov.bc.ca are considered production always so not sure if we need it

Reply:
oh yes, I was hitting the same `idsit` for all three envs while testing to prevent cluttering up there actual apis, so was changing the client name between them to prevent conflicts. I can take this out, that raises a question though in the github secrets right now I also have them set to just hit idsit for all envs. Do we want to keep that for now?

Reply:
nvm if we switch to using id.gov.bc.ca for all the environments then it would be useful for us in future.

Reply:
Oh actually testing it out you can repeat the name in the same env. So yes Ill take that out